lib/noted/notes.ex

```diff
@@ -347,8 +361,10 @@ defmodule Noted.Notes do
 
   def add_tag(user_id, note_id, tag_name) do
```

Comment:
I think the logic can be simplified:

- Get tag
- Check if tag exists, otherwise create tag, regardless, now we have tag or we return the error
- Check if notes_tag exists, otherwise create it, now we return :ok or the error

Also, when using Repo.transaction you should call Repo.rollback to return any error/failure.